pkg/controller/remotecluster/rcmgr.go

```diff
@@ -0,0 +1,132 @@
+/*
+  Copyright 2021 The Hybridnet Authors.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package remotecluster
+
+import (
+	"context"
+	"fmt"
+
+	networkingv1 "github.com/oecp/rama/pkg/apis/networking/v1"
+	"github.com/oecp/rama/pkg/rcmanager"
+	"github.com/oecp/rama/pkg/utils"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog"
+)
+
+func (c *Controller) startRemoteClusterMgr(clusterName string) error {
+	klog.Infof("processNextRemoteClusterMgr name=%v", clusterName)
+	rcManager, exists := c.rcMgrCache.Get(clusterName)
+	if !exists {
+		klog.Errorf("Can't find rcManager. clusterName=%v", clusterName)
+		return errors.Errorf("Can't find rcManager. clusterName=%v", clusterName)
+	}
+	rcManager.Run()
+	return nil
+}
+
+// use remove+add instead of update
+func (c *Controller) addOrUpdateRCMgr(rc *networkingv1.RemoteCluster) error {
+	// lock in function range to avoid renewing cluster manager when newing one
+	c.rcMgrCache.RWMutex.Lock()
```

Comment:
Maybe there is not needed anymore if using `rcMgrCache.Del()` and. `rcMgrCache.Add()`.

Reply:
If so, will there be some problem? For example, if `updateRemoteClusterStatus` is triggered by the timer when `addOrUpdateRCMgr` is called too.  The for loop in `updateRemoteClusterStatus`  will get a remote cluster manager, then `rcMgrCache.Del()` in `addOrUpdateRCMgr` is called and delete the old manager and recreate a new remote cluster manager.

Reply:
I don't think the same problem will not happen for current version of codes, because the mutex of `rcMgrCache` will not be locked any more either in `updateRemoteClusterStatus` after getting a remote cluster manager from `rcMgrCache`.

So maybe focus on the `c.rcMgrCache.RWMutex.Lock()` is not the right way to deal with your problem.

Reply:
Yes, and so far I think that getting the old remote cluster manager is not a big problem. I prefer directly using locked function and we can remove the `RWMutex` and use `sync.Map` instead.

Reply:
> Yes, and so far I think that getting the old remote cluster manager is not a big problem. I prefer directly using locked function and we can remove the `RWMutex` and use `sync.Map` instead.

OK, maybe just change to `sync.Map` then we can resolve this conversation.